pkg/devfile/generator/generators.go

```diff
@@ -55,54 +55,49 @@ func GetObjectMeta(name, namespace string, labels, annotations map[string]string
 	return objectMeta
 }
 
-// GetContainers iterates through the devfile components and returns a slice of the corresponding containers
+// GetContainers iterates through all container components, filters out init containers and returns corresponding containers
 func GetContainers(devfileObj parser.DevfileObj, options common.DevfileOptions) ([]corev1.Container, error) {
-	var containers []corev1.Container
-
-	options.ComponentOptions = common.ComponentOptions{
-		ComponentType: v1.ContainerComponentType,
-	}
-	containerComponents, err := devfileObj.Data.GetComponents(options)
+	allContainers, err := getAllContainers(devfileObj, options)
 	if err != nil {
 		return nil, err
 	}
-	for _, comp := range containerComponents {
-		envVars := convertEnvs(comp.Container.Env)
-		resourceReqs := getResourceReqs(comp)
-		ports := convertPorts(comp.Container.Endpoints)
-		containerParams := containerParams{
-			Name:         comp.Name,
-			Image:        comp.Container.Image,
-			IsPrivileged: false,
-			Command:      comp.Container.Command,
-			Args:         comp.Container.Args,
-			EnvVars:      envVars,
-			ResourceReqs: resourceReqs,
-			Ports:        ports,
+
+	// filter out init containers
+	preStartEvents := devfileObj.Data.GetEvents().PreStart
```

Comment:
how about PostStop?

PostStop are also Apply commands that can reference a component container and they are not the main containers i.e.; PostStop is executed after the main components are brought down 🤔

Reply:
postStop event is not for  init containers tho.

Reply:
but postStop are probably not part of the main containers too 🤔

Reply:
you are right. updated `GetContainers()`, and also updated the test to filter out containers for preStart and postStop events.